seqr/views/apis/report_api.py

```diff
@@ -595,6 +601,207 @@ def _get_sample_airtable_metadata(sample_ids, user, include_collaborator=False):
+        phenotype_rows += [
+            dict(**base_phenotype_row, **_get_phenotype_row(feature)) for feature in individual.features or []
+        ]
+        base_phenotype_row['presence'] = 'Absent'
+        phenotype_rows += [
+            dict(**base_phenotype_row, **_get_phenotype_row(feature)) for feature in individual.absent_features or []
+        ]
+
+        # analyte table
+        analyte_rows.append(dict(participant_id=participant_id, **_get_analyte_row(individual)))
+
+    airtable_rows = []  # TODO populate airtable data once new columns are confirmed
+
+    return export_multiple_files([
+        ['participant', PARTICIPANT_TABLE_COLUMNS, participant_rows],
+        ['family', GREGOR_FAMILY_TABLE_COLUMNS, list(family_map.values())],
+        ['phenotype', PHENOTYPE_TABLE_COLUMNS, phenotype_rows],
+        ['analyte', ANALYTE_TABLE_COLUMNS, analyte_rows],
+        ['experiment_dna_short_read', EXPERIMENT_TABLE_COLUMNS, airtable_rows],
+        ['aligned_dna_short_read', READ_TABLE_COLUMNS, airtable_rows],
+        ['aligned_dna_short_read_set', READ_SET_TABLE_COLUMNS, airtable_rows],
+        ['called_variants_dna_short_read', CALLED_TABLE_COLUMNS, airtable_rows],
+    ], f'GREGoR Reports {consent_code}', file_format='tsv')
+
+
+def _get_gregor_family_row(family):
+    return {
+        'family_id':  f'Broad_{family.family_id}',
+        'internal_project_id': f'Broad_{family.project.name}',
+        'consanguinity': 'Unknown',
+        'pmid_id': '|'.join(family.pubmed_ids or []),
+        'phenotype_description': family.coded_phenotype,
+    }
+
+
+def _get_participant_row(individual):
+    participant = {
+        'gregor_center': 'Broad',
+        'paternal_id': f'Broad_{individual.father.individual_id}' if individual.father else '0',
+        'maternal_id': f'Broad_{individual.mother.individual_id}' if individual.mother else '0',
+        'prior_testing': '|'.join([gene.get('gene', '') for gene in individual.rejected_genes or []]),
+        'proband_relationship': individual.get_proband_relationship_display(),
+        'sex': individual.get_sex_display(),
+        'affected_status': individual.get_affected_display(),
+        'reported_race': GREGOR_ANCESTRY_MAP.get(individual.population, 'Unknown'),
+        'ancestry_detail': GREGOR_ANCESTRY_DETAIL_MAP.get(individual.population),
+        'reported_ethnicity': ANCESTRY_MAP[HISPANIC] if individual.population == HISPANIC else 'Unknown',
+        'recontactable': None,  # TODO populate airtable data once new columns are confirmed
+    }
+    if individual.birth_year and individual.birth_year > 0:
+        participant.update({
+            'age_at_last_observation': str(datetime.now().year - individual.birth_year),
+            'age_at_enrollment': str(individual.created_date.year - individual.birth_year),
+        })
+    return participant
+
+
+def _get_phenotype_row(feature):
+    qualifiers_by_type = {
+        q['type']: HPO_QUALIFIERS[q['type']][q['label']]
```

Comment:
These fields are strictly enumerated so that should be an impossible case and we want this to fail spectacularly if somehow the data is malformed, we do not want this silently putting in an incorrect blank value